CMakeLists.txt

```diff
@@ -283,20 +283,41 @@ add_custom_target(hugo-deploy
     VERBATIM
 )
 
-add_custom_target(docs_dir_current
+add_custom_target(docs_source_directory
+    VERBATIM
+
+    # Clear any existing files in the target directory.
+    COMMAND ${CMAKE_COMMAND} -E remove_directory
+        ${MONGO_CXX_DRIVER_SOURCE_DIR}/build/doxygen/install
+
+    # Manually "install" all headers without requiring compilation.
+    COMMAND ${CMAKE_COMMAND} -E copy_directory
+        ${MONGO_CXX_DRIVER_SOURCE_DIR}/src/bsoncxx/include
+        ${MONGO_CXX_DRIVER_BINARY_DIR}/src/bsoncxx/lib
+        ${MONGO_CXX_DRIVER_SOURCE_DIR}/build/doxygen/install/include
+    COMMAND ${CMAKE_COMMAND} -E copy_directory
+        ${MONGO_CXX_DRIVER_SOURCE_DIR}/src/mongocxx/include
+        ${MONGO_CXX_DRIVER_BINARY_DIR}/src/mongocxx/lib
+        ${MONGO_CXX_DRIVER_SOURCE_DIR}/build/doxygen/install/include
+)
+
+add_custom_target(docs_target_directory
     COMMAND ${CMAKE_COMMAND} -E make_directory docs/api/current
 )
 
+# Set `DOXYGEN_EXECUTABLE` to specify the full path to a custom Doxygen binary.
+find_package(Doxygen 1.12.0 EXACT QUIET)
+
 add_custom_target(doxygen-current
-    DEPENDS docs_dir_current
+    DEPENDS Doxygen::doxygen docs_source_directory docs_target_directory
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    COMMAND doxygen ${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile
+    COMMAND Doxygen::doxygen
```

Comment:
Rather than using `find_package`, consider using a bash script like `generate-latest-apidocs.sh` to find and run Doxygen.

It may help make configuration more consistent. Both scripts can check `DOXYGEN_BINARY`. It may improve the error for mismatched Doxygen versions. When testing with Doxygen 1.11.0, I got this error from the `doxygen-current` target:

```
ninja: error: 'Doxygen::doxygen', needed by 'CMakeFiles/doxygen-current', missing and no known rule to make it
```

But the `doxygen-latest` target error includes the version:
```
doxygen version 1.11.0 does not equal 1.12.0
```

Reply:
Extended the `generate-latest-apidoc.sh` script to support a `--current` flag. This flag is now used by the `doxygen-current` target. Behavior is the same as without `--current` except that the _current_ repository's `build` directory is used to configure, generate, and run Doxygen without additional patching/modification of the Doxyfile.